arch/arc/core/arc_smp.c

```diff
@@ -76,6 +76,7 @@ void z_arch_slave_start(int cpu_num)
 	z_icache_setup();
 	z_irq_setup();
 
+	z_arc_connect_ici_clear();
```

Comment:
I'd say it makes no sense. We assume that we start from clean platform, otherwise we'll end-up trying to init way too many things, some of which we don't have access to or don't now about.

In case of simulation we start from clean state anyway, in case of HW platform a rule of thumb is to reset your board before execution.

Reply:
FWIW, as someone not at all expert on the platform: warm reset conditions can routinely leave hardware in crazy states at startup.  This looks safe and harmless to me.

Reply:
Yes, this part is to make code more safe.  As for hsdk and nsim,  in debug,  we just can make a warm reset.

Reply:
If we leave this I'd suggest to add a comment explaining what are we trying to achieve here and warn people about possible issues on warm reset.